packages/atomic/src/components/search/facets/atomic-automatic-facet/atomic-automatic-facet.tsx

```diff
@@ -0,0 +1,150 @@
+import {isNullOrUndefined} from '@coveo/bueno';
+import {AutomaticFacet, SearchStatus, FacetValue} from '@coveo/headless';
+import {Component, Prop, State, h, VNode} from '@stencil/core';
+import {
+  FocusTarget,
+  FocusTargetController,
+} from '../../../../utils/accessibility-utils';
+import {getFieldValueCaption} from '../../../../utils/field-utils';
+import {
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../../utils/initialization-utils';
+import {FacetContainer} from '../../../common/facets/facet-container/facet-container';
+import {FacetHeader} from '../../../common/facets/facet-header/facet-header';
+import {FacetValueCheckbox} from '../../../common/facets/facet-value-checkbox/facet-value-checkbox';
+import {FacetValueLabelHighlight} from '../../../common/facets/facet-value-label-highlight/facet-value-label-highlight';
+import {FacetValuesGroup} from '../../../common/facets/facet-values-group/facet-values-group';
+import {Hidden} from '../../../common/hidden';
+import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
+
+/**
+ * An automatic facet is a facet generated by the automatic facets feature. This component should
```

Comment:
I did a design doc 1 month ago. The part about atomic/headless is a bit outdated but the context/searchAPI part is the same. there is also a bunch of links in the epic if you are curious. Basically, right now we are sending requests for every facets. Some clients have thousands of facets. We would like to have facets generated by the index that are dynamic and that depend on the search query. 

There is no mention on the doc site because this is a beta search api feature. From what I understand, there is only 1 client using this feature (Bunnings). The GA will probably be in september.

I will add a sentence to clear up the difference between normal and automatic facets. 

https://coveord.atlassian.net/browse/KIT-2503
https://docs.google.com/document/d/1E6OdgIdxovs2_wcTyemsKNtOXU7N9VTP9CpgKPacyho/edit#heading=h.uyanedjmkl4o